pkg/util/net/tcpsock.go

```diff
@@ -0,0 +1,34 @@
+package net
+
+import (
+	"fmt"
+	"net"
+)
+
+func PickTCPPort(ip string, leftPort, rightPort uint32) (actualPort uint32, err error) {
+	lowestPort, highestPort := leftPort, rightPort
+	if highestPort < lowestPort {
+		lowestPort, highestPort = highestPort, lowestPort
+	}
+	// we prefer a port to remain stable over time, that's why we do sequential availability check
+	// instead of random selection
```

Comment:
What is the use case to have it stable?

Reply:
It's probably opinionated.

When, as a user, I define a port range `9901-9999`, I expect that application will choose the first available port.